libraries/Microsoft.Bot.Builder.Ai/QnAMakerMiddleware.cs

```diff
@@ -12,10 +13,23 @@ namespace Microsoft.Bot.Builder.Ai
     public class QnAMakerMiddleware : Middleware.IReceiveActivity
     {
         private readonly QnAMaker _qnaMaker;
+        private readonly QnAMakerMiddlewareOptions _options;
 
-        public QnAMakerMiddleware(QnAMakerOptions options, HttpClient httpClient)
+        public QnAMakerMiddleware(QnAMakerMiddlewareOptions options, HttpClient httpClient)
         {
-            _qnaMaker = new QnAMaker(options, httpClient);
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+
+            var qnaMakerOptions = new QnAMakerOptions()
```

Comment:
I'm torn on this pattern. I get it, but it may be overkill here. Time will tell if it's applicable to other (similar) middlware + library modules.

Reply:
I do see your point, but I can certainly see potential in this case for the number of optional params within the middleware to increase overtime as it becomes a little more flexible.  I think it works here, but agree that it won't be appropriate everywhere.